fix: Skip meshery-integration-template model during registry generate

Ontiomacer opened the pull request:
This PR fixes meshery/meshery #14702.

In the Meshery Integrations Spreadsheet, the row
meshery-integration-template is an example/template model used for
creating new model entries. It should not be processed during
mesheryctl registry generate.

This PR adds a simple conditional check inside registry/model.go to
skip this model gracefully, preventing the generator from attempting to
register a template model and avoiding unnecessary errors in the logs.
--------------------------------------------------------------------------------------
- Added conditional to skip model == "meshery-integration-template"
- Prevents generator from processing a template-only entry
- Eliminates the "generator not implemented" error during registry generate
---------------------------------------------------------------------------------------
What I changed :
- Added a conditional in registry/model.go to skip the model named meshery-integration-template.
- Prevents the registry generator from processing the template/example row.
- Removes the unnecessary “generator not implemented” errors during mesheryctl registry generate.
Ran:
mesheryctl registry generate --directory ./mesheryctl/templates/template-csvs

Ontiomacer commented:
@leecalcote This PR resolves meshery/meshery#14702.

It adds a constant for meshery-integration-template and ensures this
template-only model is skipped during registry generation. Verified that
normal models continue to generate correctly.

marblom007 commented on registry/model.go:
```diff
@@ -838,10 +837,14 @@ func InvokeGenerationFromSheet(wg *sync.WaitGroup, path string, modelsheetID, co
 	// Iterate models from the spreadsheet
 	for _, model := range modelCSVHelper.Models {
-
+		if model.Model == mesheryIntegrationTemplateModel{
```
Matching on this model name makes for brittle code. Can we move to only processing components that have a `true` value for `PublishToRegistry`, instead?

Ontiomacer commented:
Thanks, @leecalcote — understood.

I’ll update this PR to implement the correct approach:
• Only process models where PublishToRegistry == true
• Ensure the template row and any other non-publishable rows are naturally excluded

I’ll also take a look at the nightly generator workflow errors and verify that the updated logic resolves them.

Will push the updated commit shortly.

Ontiomacer commented:
Update: Implemented PublishToRegistry-based filtering
Thanks for the feedback, everyone — especially @marblom007, @lekaf974, and @leecalcote.
I've updated the PR to follow the recommended approach:
✔ What changed
Removed the previous model-name–based skip.
The generator now processes only models where PublishToRegistry == true, exactly as suggested.
When a specific model is passed using --model, publish checks are bypassed so users can always generate for that model.
Added an informational log message when a model is skipped due to PublishToRegistry != true.
✔ Result
This removes the brittle string match and aligns the generator with the intended spreadsheet-driven behavior.
Please let me know if you'd like any additional refinements!
